### PR TITLE
 Add option to uninstall all clusters

### PR DESCRIPTION
**Pull request by rnetser**

### CODE REVIEW

**Review by omrirh — APPROVED** [review]

**Review by oharan2 — APPROVED** [review]

File: app/cli.py [hunk 1]
```diff
@@ -253,11 +291,35 @@ def main(
...
+    if destroy_all_clusters or destroy_clusters_from_config_files:
+        destroy_clusters(
+            s3_bucket_name=s3_bucket_name,
+            s3_bucket_path=s3_bucket_path,
+            clusters_install_data_directory=clusters_install_data_directory,
+            registry_config_file=registry_config_file,
+            clusters_yaml_files=destroy_clusters_from_config_files,
+            destroy_all_clusters=destroy_all_clusters,
+        )
+        return
```
Comment (myakove) [inline app/cli.py:315]: better to `return destroy_clusters`, move the return to line 307

File: app/libs/destroy_clusters.py [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+            extract_dir=target_extract_dir,
+            format="zip",
+        )
+    except client.exceptions.ClientError as ex:
+        if ex.response["Error"]["Code"] == str(HTTPStatus.NOT_FOUND):
+            click.echo(f"{bucket_filepath} not found in {bucket}")
+            raise click.Abort()
+
+
+def _destroy_clusters_download_installer_binary(
```
Comment (myakove) [inline app/libs/destroy_clusters.py:46]: The name is confusing, when reading this I this you will destroy the binaries directories.

File: app/libs/destroy_clusters.py [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+    if aws_clusters:
+        download_openshift_install_binary(
+            clusters=aws_clusters, registry_config_file=registry_config_file
+        )
+
+
+def destroy_clusters_from_data_dict(cluster_data_dict, s3_bucket_name=None):
+    processes = []
+
+    for cluster_type in cluster_data_dict:
```
Comment (myakove) [inline app/libs/destroy_clusters.py:59]: `cluster_data_dict.items()` ?

File: app/libs/destroy_clusters.py [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+
+    for proc in processes:
+        proc.join()
+
+
+def _destroy_cluster(cluster_data, cluster_type, s3_bucket_name=None):
+    try:
+        if cluster_type == AWS_STR:
+            create_or_destroy_aws_ipi_cluster(
+                cluster_data=cluster_data, action="destroy"
```
Comment (myakove) [inline app/libs/destroy_clusters.py:81]: `"destroy"` should be constant

File: app/libs/destroy_clusters.py [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+    click.echo(f"Prepare target extracted directory {extracted_target_dir}.")
+    Path(extracted_target_dir).mkdir(parents=True, exist_ok=True)
+    return extracted_target_dir, target_dir
+
+
+def get_clusters_data(cluster_dirs, clusters_dict):
+    def _get_cluster_dict_from_yaml(_cluster_filepath):
+        with open(_cluster_filepath) as fd:
+            _data = yaml.safe_load(fd.read())
+        _data["install-dir"] = root
```
Comment (myakove) [inline app/libs/destroy_clusters.py:117]: you should get `root` as `_root` in the inner function

**Review by myakove — CHANGES_REQUESTED** [review]

**Review by myakove — APPROVED** [review]

**Pull request merged**